src/odemis/acq/move.py

```
@@ -1305,11 +1313,11 @@ def _do_cryo_switch_sample_position(self, future, target):
         """
         Do the actual switching procedure for cryo_switch_sample_position
         :param future: cancellable future of the move
-        :param target: (int) target position either one of the constants: LOADING, SEM_IMAGING, FM_IMAGING.
+        :param target: (int) target position either one of the constants: Posture.LOADING, Posture.SEM_IMAGING, Posture.FM_IMAGING.
         """
         try:
             try:
-                target_name = POSITION_NAMES[target]
+                target_name = target.value
             except KeyError:
                 raise ValueError(f"Unknown target '{target}'")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)